src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/multiblocks/TableSaw.java

```diff
@@ -54,28 +64,44 @@ public TableSaw(Category category, SlimefunItemStack item) {
     }
 
     @Override
-    public void onInteract(Player p, Block b) {
-        ItemStack log = p.getInventory().getItemInMainHand();
-
-        Optional<Material> planks = MaterialConverter.getPlanksFromLog(log.getType());
+    public void onInteract(@Nonnull Player p, @Nonnull Block b) {
+        ItemStack item = p.getInventory().getItemInMainHand();
+        ItemStack output = getItemsToOutput(item);
 
-        if (planks.isPresent()) {
-            if (p.getGameMode() != GameMode.CREATIVE) {
-                ItemUtils.consumeItem(log, true);
-            }
+        if (output == null) {
+            return;
+        }
 
-            ItemStack output = new ItemStack(planks.get(), 8);
-            Inventory outputChest = findOutputChest(b, output);
+        if (p.getGameMode() != GameMode.CREATIVE) {
+            ItemUtils.consumeItem(item, true);
+        }
 
-            if (outputChest != null) {
-                outputChest.addItem(output);
-            }
-            else {
-                b.getWorld().dropItemNaturally(b.getLocation(), output);
-            }
+        outputItems(b, output);
+        b.getWorld().playEffect(b.getLocation(), Effect.STEP_SOUND, item.getType());
+    }
 
-            b.getWorld().playEffect(b.getLocation(), Effect.STEP_SOUND, log.getType());
+    @Nullable
+    private ItemStack getItemsToOutput(@Nonnull ItemStack item) {
```

Comment:
Instead of `ItemStack item` you can use `Material type` to reduce the calls to get the type.